src/keystrokehandler.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import EmitterMixin from './utils/emittermixin.js';
+import KeyObserver from './engine/view/observer/keyobserver.js';
+import { getCode, parseKeystroke } from './utils/keyboard.js';
+
+/**
+ * Keystroke handler. Its instance is available in {@link ckeditor5.Editor#keystrokes} so features
+ * can register their keystrokes.
+ *
+ * E.g. an undo feature would do this:
+ *
+ *		editor.keystrokes.set( 'ctrl + Z', 'undo' );
+ *		editor.keystrokes.set( 'ctrl + shift + Z', 'redo' );
+ *		editor.keystrokes.set( 'ctrl + Y', 'redo' );
+ *
+ * @memberOf ckeditor5
+ */
+export default class KeystrokeHandler {
+	/**
+	 * Creates an instance of the keystroke handler.
+	 *
+	 * @param {engine.treeView.TreeView} editingView
+	 */
+	constructor( editor ) {
+		/**
+		 * The editor instance.
+		 *
+		 * @readonly
+		 * @member {ckeditor5.Editor} ckeditor5.KeystrokeHandler#editor
+		 */
+		this.editor = editor;
+
+		/**
+		 * Observer used to listen to events for easier keystroke handler destruction.
+		 *
+		 * @private
+		 * @member {utils.Emitter} ckeditor5.KeystrokeHandler#_observer
+		 */
+		this._observer = Object.create( EmitterMixin );
```

Comment:
But, in fact, any name is fine for me, as long we do not have a naming conflict. When I saw `_observer`, I was sure that it is an `KeyObserver` used by `KeystrokeHandler`. So I am fine with `_listeners`, `_subscribers` or renaming `Observer` class to `Listener` (in fact most of them are DOM listeners), but I do not like using same name for both meanings.